common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -88,12 +88,9 @@ public static void submitSilent(@NonNull final BaseCommand command) {
         );
 
         if (sExecutingCommands.contains(command)) {
```

Comment:
OK, what would make this a little more readable, and flexible, is to cache the Future<Result> of this command.  We might want to look into importing streamsupport-cfuture or doing our own back port of CompletableFuture using androidx's SettableFuture as the basis if the GPLv2\Classpath license isn't acceptable.  This will give us a class that we can use for this kind of behavior in multiple places.

Reply:
The main thing that would change would be sExecutingCommands would change from a Set to a Map<BaseCommand, CompletableFuture<CommandResult>>.  Essentially what this becomes is a Cache that can store incomplete values.

Reply:
Going to close this PR, will prep another with discussed changes